### PR TITLE
GraphQL - Queries

### PR DESCRIPTION
**Pull request by Aurelsicoko**
This PR is the first step of the GraphQL integration. It only supports query, not mutation yet.

### CODE REVIEW

File: packages/strapi-plugin-content-manager/config/layout.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "attributes": {
+      "title": {
+        "appearance": ""
+      },
+      "content": {
+        "appearance": "WYSIWYG"
+      }
+    }
+  }
+}
```
Comment (lauriejim) [inline packages/strapi-plugin-content-manager/config/layout.json]: File to remove or add to .gitignore

File: packages/strapi-plugin-graphql/package.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    "url": ""
+  },
+  "maintainers": [
+    {
+      "name": "A Strapi developer",
+      "email": "",
+      "url": ""
+    }
+  ],
+  "engines": {
```
Comment (lauriejim) [inline packages/strapi-plugin-graphql/package.json]: Update engine versions

File: packages/strapi-plugin-graphql/services/GraphQL.js [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+
+                return entry[association.alias];
+              }
+            });
+            break;
+          default:
+        }
+
+        // TODO:
+        // - Handle limit, skip, etc options
```
Comment (lauriejim) [inline packages/strapi-plugin-graphql/services/GraphQL.js]: Remove this todo, it's done

File: packages/strapi-plugin-upload/config/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  resolver: {
+    Query: {
+      file: false,
+      files: {
+        resolver: 'Upload.find'
+      }
+    }
+  }
+};
```
Comment (lauriejim) [inline packages/strapi-plugin-upload/config/schema.graphql]: File to delete ? Or let it for examples.

File: packages/strapi/lib/hooks/index.js [hunk 5]
```diff
@@ -63,44 +83,61 @@ module.exports = function() {
             defaultsDeep(this.config.hook.settings[hook], module(this).defaults[hook] || module(this).defaults);
           }
 
-          // Take care of hooks internals dependencies.
-          if (dependencies.length > 0 || includes(get(this.config.hook, 'load.order', []), hook)) {
-            const position = indexOf(get(this.config.hook, 'load.order', []), hook);
-            const previousDependencies = dropRight(get(this.config.hook, 'load.order', []), get(this.config.hook, 'load.order', []).length - (position + 1));
+          // Initialize array.
+          let previousDependencies = this.hook[hook].dependencies.map(x => x.replace('strapi-', '')) || [];;
```
Comment (lauriejim) [inline packages/strapi/lib/hooks/index.js]: Double `;` at the end of the line

File: packages/strapi-plugin-upload/config/schema.graphql (hunk 4 above)
Comment (Aurelsicoko) [reply packages/strapi-plugin-upload/config/schema.graphql]: No, we need it to make GraphQL working with the Upload plugin.

**Comment by richardscarrott** [comment]
Do you know when graphql mutations are likely to land if at all?

**Comment by Aurelsicoko** [comment]
@richardscarrott At the end of August or maybe before... I need to find some time to continue the work on this part and I don't have any right now :(

**Pull request merged**